### ISSUE TITLE
Incorrect typescript types for parameters using dot notation

### ISSUE THREAD

**Issue by kristofferso** [QuiiBz/next-international#197]
**Describe the bug**
Using dot notation for a translation in a Typescript file with parameters ruins the type for the params argument in the `t()` function. Using object notation works fine. This results in the error `Expected 1 arguments, but got 2` from Typescript, which is inferring the parameter types to be `never`.

**To Reproduce**
1. Add a translation using a dot notation from your default language translation file (only tested with .ts), with a parameter. Example: `export default {'welcome.user': 'Hello {name}!'} as const`
2. Use the translation in a React Server Component like a `page.tsx`: 
```
export default async function Page(){
const t = await getI18n();
... 
return <p>{t("welcome.user", {name: "Phil"})}</p>
}
```

**Expected behavior**
Expected types for the second argument of `t()` to show parameter types. Instead it is inferred as `never`. 

**About:**
 - next-international version: 1.0.1
 - Next.js version: 13.5.1

**Comment by QuiiBz** [comment]
I'm not able to reproduce the issue using the example: https://github.com/QuiiBz/next-international/tree/main/examples/next-app

![Screenshot 2023-09-28 at 08 29 01](https://github.com/QuiiBz/next-international/assets/43268759/89514add-f283-4ce8-ba64-1f2e76aa402f)

Have you set `strict: true` inside your `tsconfig.json` file? You could also try to upgrade your TypeScript version. Otherwise, I'll need a reproduction to check why you're getting this error.

**Comment by QuiiBz** [comment]
You have to build `packages/next-international` and `packages/international-types` before running the example. You can also check the codesandbox here that doesn't have the issue: https://codesandbox.io/p/sandbox/jovial-paper-skkprk?file=%2Fapp%2F%5Blocale%5D%2Fpage.tsx%3A1%2C1

**Comment by gustaveWPM** [comment]
ping @kristofferso ?

**Comment by gustaveWPM** [comment]
Maybe related: https://github.com/QuiiBz/next-international/issues/231

**Comment by QuiiBz** [comment]
We'll have to see your `createI18nClient` / `createI18nServer` methods to see if it's related to #231 or not. If no response in the next few days, I'll go ahead and close the issue.

**Comment by kristofferso** [comment]
Sorry about the late response here :( 

This is my client.ts: 
```
import { createI18nClient } from 'next-international/client';

export const {
  useI18n,
  useScopedI18n,
  I18nProviderClient,
  useChangeLocale,
  useCurrentLocale,
} = createI18nClient({
  no: () => import('./no'),
  se: () => import('./se'),
});
```

server.ts
```
import { createI18nServer } from 'next-international/server';

export const { getI18n, getScopedI18n, getStaticParams, getCurrentLocale } =
  createI18nServer({
    no: () => import('./no'),
    se: () => import('./se'),
  });
```
I'm not doing anything special as I can see.

**Comment by kristofferso** [comment]
And here is the error: 
```
Expected 1 arguments, but got 2.ts(2554)
⚠ Error(TS2554)  | 
Expected 1 arguments, but got 2.
```
<img width="734" alt="bilde" src="https://github.com/QuiiBz/next-international/assets/42342956/82572489-37b3-46c8-be59-ae00af156873">

I'll do some more testing with the example repo

**Comment by kristofferso** [comment]
So I was able to recreate this in the codesandbox example. Just add any type of nesting to the locale file together with a parameter string and you get the error. 

Example: 
```
export default {
  hello: "Hello",
  "hello.world": "Hello world!",
  "hello.world.welcome": "Hello {name}!",
  welcome: "Hello {name}!",
  test: {
    anything: "all at once",
  },
} as const;
```

**Comment by QuiiBz** [comment]
Thanks, I found that it's because you're mixing dot and object notation at the same time. This sounds like a bug that shouldn't happen so I'll fix it asap, but in the meantime you can use only dot notation or only object notation.

**Comment by QuiiBz** [comment]
The fix (#236) will be released soon.

**Comment by QuiiBz** [comment]
Released in [next-international@1.1.2](https://github.com/QuiiBz/next-international/releases/tag/1.1.2)!

**Comment by kristofferso** [comment]
Very nice 👏
